sdk/python/kfp/cli/pipeline.py

```diff
@@ -57,10 +63,16 @@ def upload(ctx, pipeline_name, package_file):
 )
 @click.argument("package-file")
 @click.pass_context
-def upload_version(ctx, package_file, pipeline_version, pipeline_id):
+def upload_version(ctx, package_file, pipeline_version, pipeline_id=None, pipeline_name=None):
     """Upload a version of the KFP pipeline"""
     client = ctx.obj["client"]
-
+    if (pipeline_id == None and pipeline_name == None)  or (pipeline_id and pipeline_name): #[TODO] do this using click instead... 
+        raise ValueError("Need to suppy 'pipeline-name' or 'pipeline-id'")
+    if pipeline_name!=None: 
+        pipeline_id = client.get_pipeline_id(name=pipeline_name)
+        if pipeline_id==None: 
+            raise ValueError("Cant find a pipeline with name: {}".format(pipeline_name))
+    logging.info("The pipeline id is: {}".format(pipeline_id))
```

Comment:
Consider using % formatter in logging when possible. This is b/c if we use `format`, the string template gets resolved even when not needed.

See explanation in
https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html
for example.

Reply:
Didn't know, thanks :) I will update them!